### PR TITLE
Move smoke test code to Skupper org

### PR DESCRIPTION
**Pull request by granzoto**
Migration to Skupper org

### CODE REVIEW

File: src/main/main.go [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+
+	skupper_ocp_smoke.PrintIfDebug("main : Kubeconfig for public namespace = ", os.Getenv("PUBKUBECONFIG"))
+	skupper_ocp_smoke.PrintIfDebug("main : Kubeconfig for private namespace = ", os.Getenv("PRIVKUBECONFIG"))
+
+	// Context and CLI
+	ctx, cn := context.WithTimeout(context.Background(), time.Minute*30)
+	defer cn()
+
+	pubCli, _ := skupper_ocp_smoke.NewClient("", pubKubeconfigFile, skupper_ocp_smoke.PUBNS)
+	privCli, _ := skupper_ocp_smoke.NewClient("", privKubeconfigFile, skupper_ocp_smoke.PRIVNS)
```
Comment (fgiorgetti) [inline src/main/main.go]: The `NewClient` function is returning a nil value to me.
It is better to check/log eventual errors returned here.

Comment (granzoto) [reply src/main/main.go]: Added an error checking

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
+FROM golang:1.19 AS builder
+
+WORKDIR /image
+COPY src/go.mod .
+COPY src/go.sum .
+RUN go mod download
+COPY . .
```
Comment (hash-d) [inline Dockerfile:4]: Why do src/go.{mod,sum} need to go on /image and not on /image/src?  Working with them on their original location would remove these two copy commands.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,13 @@
+VERSION := $(shell git describe --tags --dirty=-modified --always)
+SMOKE_TEST_IMAGE := quay.io/skupper/skupper-ocp-smoke-test-image
+DOCKER := docker
+
+all: build-smoke-image push-smoke-image
```
Comment (hash-d) [inline Makefile]: The `all` target is used when `make` is called without arguments.

Having it push to quay may cause some unintended pushes.  I'd use something else, like `everything`, so the will to make and push is always explicit

File: README.md [hunk 4]
```diff
@@ -1 +1,144 @@
...
+
+  After the Skupper instances are up, the test image connects them, setup a simple web server in the priv-test-ns namespace, and then access this service through Skupper in the pub-test-ns namespace.
+
+<br>
+
+## Basic usage :
+
+**docker**:
+
+    "docker run -v /home/localuser/kubeconfigs/:/config  -it --env PUBKUBECONFIG=/config/config-ocp49 --env PRIVKUBECONFIG=/config/config-ocp410 --env QUIET=TRUE quay.io/skupper/skupper-ocp-smoke-test-image"
```
Comment (hash-d) [inline README.md]: When the MarkDown is rendered into HTML by GitHub, the quotes at start and end are being show, and they're copied when one clicks the copy to clipboard button.   That would be an incorrect command.  Please consider removing these quotation marks.

File: README.md [hunk 5]
```diff
@@ -1 +1,144 @@
...
+**Openshift**:
+
+    "oc create -f skupper-ocp-smoke-test.yaml -n [namespace]"
+
+<br>
+
+Openshift
+==========
+
+ Use the script generate-yaml.sh to generate the YAMl file to run the Skupper Openshift smoke tests.
```
Comment (hash-d) [inline README.md]: Small typo: `YAMl` vs `YAML`

File: README.md [hunk 6]
```diff
@@ -1 +1,144 @@
...
+
+<br>
+
+## Advanced configuration
+
+  The test runs in three steps : Setup, Test and Teardown. If you don't specify a STEP, the test will go through all of them.
+
+  Usually you don't need to specify a step, but if you need, you can do it using the STEP environment variable :
+
+    "-- env STEP=SETUP"<br>
```
Comment (hash-d) [inline README.md]: This `<br>` is being shown verbatim on the rendered output

File: README.md [hunk 7]
```diff
@@ -1 +1,144 @@
...
+    ** This will create the namespaces, deploy the Skupper instances and the services, but will not validate the test. It can be used for debugging purposes.
```
Comment (hash-d) [inline README.md]: Having these long lines as code-like (indented) makes it hard to read on GitHub, as they are shown with horizontal scroll bars; perhaps just un-indent them?

**Review by hash-d — APPROVED** [review]
I haven't reviewed the code in detail, as we have already done this before and I know you've been running it, already.   I only really checked the documentation, and made some small suggestions for improvement

File: src/main/main.go [hunk 8]
```diff
@@ -0,0 +1,373 @@
...
+	log.Printf("%s : Create Subscriptions", step)
+	for _, cli := range cliList {
+		if err := cli.CreateSubscription(step); err != nil {
+			return fmt.Errorf("%s : %s", step, err.Error())
+		}
+
+		// Wait until the operator is up
+		if err := cli.WaitPodsStatus(ctx, "Running", cli.WaitLimit(), "skupper-site-controller", step); err != nil {
+			return fmt.Errorf("%s : %s", step, err.Error())
+		}
```
Comment (fgiorgetti) [inline src/main/main.go]: I would move this step after the for loop.

This is causing the 1st namespace subscription to be created, but
the 2nd is delayed till ctrl is runningon 1st namespace.

File: src/main/main.go [hunk 9]
```diff
@@ -0,0 +1,373 @@
...
+	if err := pubCli.IsSkupperOperatorAvailable(step, pubCli.OperatorName(), pubCli.OperatorNameSpace(), pubCli.OperatorCatalog()); err != nil {
+		return fmt.Errorf("%s : %s", step, err.Error())
+	}
+
+	// Create the Catalog Groups
+	log.Printf("%s : Create OperatorGroups", step)
+	for _, cli := range cliList {
+		// Create the operatorgroup
+		if err := cli.CreateOperatorGroup(step); err != nil {
+			return fmt.Errorf("%s : %s", step, err.Error())
```
Comment (fgiorgetti) [inline src/main/main.go]: It is failing here when I am using CLEANBEFORE=TRUE. I believe it should simply
ignore the "already exists" error in such case.

File: Dockerfile (hunk 2 above)
Comment (granzoto) [reply Dockerfile:4]: Agreed. Fixed

File: src/main/main.go (hunk 8 above)
Comment (granzoto) [reply src/main/main.go]: Agrred. moved to another step

File: src/main/main.go (hunk 9 above)
Comment (granzoto) [reply src/main/main.go]: Fixed for all the deployment steps, it will fail only if the Job definition already exists

File: Makefile (hunk 3 above)
Comment (granzoto) [reply Makefile]: Agreed. Split options and added some info message

File: README.md (hunk 7 above)
Comment (granzoto) [reply README.md]: I have wrapped the lines

File: README.md (hunk 4 above)
Comment (granzoto) [reply README.md]: Agreed. Removed

**Comment by ajssmith** [comment]
In looking at the project, might it be worth considering following the common project layout for golang? While not a standard, it is a set of conventions that projects seem to follow https://github.com/golang-standards/project-layout

**Comment by granzoto** [comment]
> In looking at the project, might it be worth considering following the common project layout for golang? While not a standard, it is a set of conventions that projects seem to follow https://github.com/golang-standards/project-layout

@ajssmith I have refactored the code according to what I understood from the layout standards. If you have any suggestion, please feel free.

File: cmd/main.go [hunk 10]
```diff
@@ -0,0 +1,380 @@
...
+		ObjectMeta: v12.ObjectMeta{
+			Name:      "skupper-site",
+			Namespace: pubCli.Namespace,
+		},
+		Immutable: nil,
+		Data: map[string]string{
+			"router-mode":      "interior",
+			"console-user":     "admin",
+			"console-password": "changeme",
+		},
```
Comment (fgiorgetti) [inline cmd/main.go:108]: There is one more issue here. When you run this job on OCP with a Security Context Constraint (scc)
that enforces an uid-range to be provided, the test will fail.

To bypass that I'd suggest you to use something like the Skupper's CLI tests are using, based on the
[getRunAsUserOrDefault](https://github.com/skupperproject/skupper/blob/6a0acae942d8ccb6fe60147196367289c174b761/test/integration/examples/custom/helloworld/main_test.go#L29) function.

With that said, you will need to add two more entries here:
"run-as-user" and "run-as-group", using the recommended ID range.

File: cmd/main.go [hunk 11]
```diff
@@ -0,0 +1,380 @@
...
+		ObjectMeta: v12.ObjectMeta{
+			Name:      "skupper-site",
+			Namespace: privCli.Namespace,
+		},
+		Immutable: nil,
+		Data: map[string]string{
+			"router-mode":      "interior",
+			"console-user":     "admin",
+			"console-password": "changeme",
```
Comment (fgiorgetti) [inline cmd/main.go:135]: And the same thing here... Add the "run-as-user" and "run-as-group" entries.

**Review by fgiorgetti — APPROVED** [review]
LGTM!

File: cmd/main.go (hunk 10 above)
Comment (granzoto) [reply cmd/main.go:108]: ok, added missing configs

**Pull request merged**